src/java.base/share/classes/sun/security/provider/SecureRandom.java

```diff
@@ -190,25 +213,77 @@ private static void updateState(byte[] state, byte[] output) {
         }
     }
 
+    private void invalidate() {
```

Comment:
I would like to have "assert objLock.isHeldByCurrentThread()" or another "objLock.lock()" here. This method is called from SeederHolder.beforeCheckpoint. There is no race here only if that executes after seeder's beforeCheckpoint and in the same thread. While it is so, additional safety would not harm.